Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -23,177 +23,50 @@ public class ClassificationTests: XCTestCase {
-    }
-    do {
-      let pattern = (tree.statements[0].item.as(VariableDeclSyntax.self)!).bindings[0].pattern
-      XCTAssertEqual(pattern.description, "x/*yo*/ ")
-      // Classify with a relative range inside this node.
-      let classif = Array(pattern.classifications(in: ByteSourceRange(offset: 5, length: 2)))
-      XCTAssertEqual(classif.count, 2)
-      guard classif.count == 2 else {
-        return
-      }
-      XCTAssertEqual(classif[0].kind, .blockComment)
-      XCTAssertEqual(classif[0].range, ByteSourceRange(offset: 14, length: 6))
-      XCTAssertEqual(classif[1].kind, .none)
-      XCTAssertEqual(classif[1].range, ByteSourceRange(offset: 20, length: 1))
-
-      do {
-        let singleClassif = pattern.classification(at: 5)
-        XCTAssertEqual(singleClassif, classif[0])
-      }
-      do {
-        let singleClassif = pattern.classification(at: AbsolutePosition(utf8Offset: 19))
-        XCTAssertEqual(singleClassif, classif[0])
-      }
-    }
-
-    do {
-      let source = "func foo() {}"
-      let tree = Parser.parse(source: source)
-      // For `classification(at:)` there's an initial walk to find the token that
-      // the offset is contained in and the classified ranges are processed from that
-      // token. That means that a `none` classified range would be restricted inside
-      // the token range.
-      let classif = tree.classification(at: 11)!
-      XCTAssertEqual(classif.kind, .none)
-      XCTAssertEqual(classif.range, ByteSourceRange(offset: 11, length: 1))
-    }
-  }
-
-  public func testTokenClassification() {
-    let source = "let x: Int"
-    let tree = Parser.parse(source: source)
-    do {
-      let tokens = Array(tree.tokens(viewMode: .sourceAccurate))
-      XCTAssertEqual(tokens.count, 5)
-      guard tokens.count == 5 else {
-        return
-      }
-      let classif = tokens.map { $0.tokenClassification }
-      XCTAssertEqual(classif[0].kind, .keyword)
-      XCTAssertEqual(classif[0].range, ByteSourceRange(offset: 0, length: 3))
-      XCTAssertEqual(classif[1].kind, .identifier)
-      XCTAssertEqual(classif[1].range, ByteSourceRange(offset: 4, length: 1))
-      XCTAssertEqual(classif[2].kind, .none)
-      XCTAssertEqual(classif[2].range, ByteSourceRange(offset: 5, length: 1))
-      XCTAssertEqual(classif[3].kind, .typeIdentifier)
-      XCTAssertEqual(classif[3].range, ByteSourceRange(offset: 7, length: 3))
-      XCTAssertEqual(classif[4].kind, .none)
-      XCTAssertEqual(classif[4].range, ByteSourceRange(offset: 10, length: 0))
-    }
-    do {
-      let tok = tree.lastToken(viewMode: .sourceAccurate)!.previousToken(viewMode: .sourceAccurate)!
-      XCTAssertEqual("\(tok)", "Int")
-      let classif = Array(tok.classifications).first!
-      XCTAssertEqual(classif.kind, .typeIdentifier)
-    }
+    
+    assertClassification(
+    source, expectedClassificationSpec: [
+      .init(source: "// blah.", kind: .lineComment),
```

Comment:
I generally prefer to spell this as `ClassificaitonSpec`. Then you’re not left wondering what object you’re constructing here while reading the code.